packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -36,6 +36,7 @@ export interface SdkEmitterOptions {
   "package-name"?: string;
   "flatten-union-as-enum"?: boolean;
   "api-version"?: string;
+  "flavor"?: "azure";
```

Comment:
since it is an emitter behavior to filtering preview version, i think we could add an option when create sdk context which could infer how to do preview filtering with regex. that could be more flexible i believe.

Reply:
what are your suggestions? I'm not sure I'm following

Reply:
i suggest to add an option [here](https://github.com/Azure/typespec-azure/blob/0b1afda6a4531f1e0e458e34538958f890129810/packages/typespec-client-generator-core/src/decorators.ts#L580). the option is a regex for filtering preview version.

Reply:
Maybe we need to define
```
interface CreateSdkContextOptions {
  readonly versionStrategy?: {
    strategy: "ignore|filter",
    filterFunc: (defaultApiversion: string, versions: VersionEnum[]) => void
  }
}
```

Reply:
i think that could work, and we could have the default regex value be `"-preview"`

Reply:
@chunyu3 that would be a breaking change, but I do like the idea of grouping the versioning strategy together. I looked at it seems no one is using this option. I'm going to go ahead and try to follow your suggestion, and people can leave comments on it